pynicotine/slskproto.py

```diff
@@ -1759,6 +1787,71 @@ def _process_file_output(self, msg_obj):
 
     """ Distributed Connection """
 
+    def _accept_child_peer_connection(self, conn_obj):
+
+        if conn_obj.init.conn_type != ConnectionType.DISTRIBUTED:
+            return
+
+        user = conn_obj.init.target_user
+
+        if self._parent_socket is None and not self._is_server_parent:
+            # We have no parent user and the server hasn't sent search requests, no point
```

Comment:
log.add_conn(("Rejecting distributed child peer connection from user %(user)s, since we have no parent", {"user": user))